### PR TITLE
[Janitor] more listLength

### PR DESCRIPTION
**Pull request by hkiel**
- avoid duplicate listLength()
- use listEmpty if applicable

### CODE REVIEW

File: OMCompiler/Compiler/BackEnd/ResolveLoops.mo [hunk 1]
```diff
@@ -1594,17 +1594,15 @@ protected function priorizeEqsWithVarCrosses2
...
-  numCrossVars := listLength(crossVars);
-  if numCrossVars == 0 then
+  if listEmpty(crossVars) then
     arrayGetAppendLst(1,{eq},priorities);
-  elseif numCrossVars == 1 then
+  elseif listLength(crossVars) == 1 then
     arrayGetAppendLst(2,{eq},priorities);
-  elseif numCrossVars >= 2 then
+  else
     arrayGetAppendLst(3,{eq},priorities);
```
Comment (perost) [inline OMCompiler/Compiler/BackEnd/ResolveLoops.mo:1606]: Doing `listLength` on an empty list is pretty much the same as doing `listEmpty`, so there's not much point in first checking if the list is empty here. But it probably doesn't really matter one way or another.

Comment (perost) [reply OMCompiler/Compiler/BackEnd/ResolveLoops.mo:1606]: Yes, I didn't mention that but we already have `List.hasOneElement` that you could use here.

Comment (perost) [reply OMCompiler/Compiler/BackEnd/ResolveLoops.mo:1606]: Up to you, it doesn't really matter.

Comment (hkiel) [reply OMCompiler/Compiler/BackEnd/ResolveLoops.mo]: Ok. I replaced this `listLength() == 1` by `List.hasOneElement()`. I will open a second PR with the other changes.

**Pull request merged**